### PR TITLE
Component functionality

### PR DESCRIPTION
**Pull request by glenn-operant**

### CODE REVIEW

File: internal/components/components.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+							Image:           component.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 8080,
+								},
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name: "OPEN_API_KEY",
```
Comment (priyanka-operant) [inline internal/components/components.go:131]: OPEN_AI_KEY 😆

File: internal/components/components.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+									ValueFrom: &corev1.EnvVarSource{
+										SecretKeyRef: &corev1.SecretKeySelector{
+											LocalObjectReference: corev1.LocalObjectReference{
+												Name: component.Type,
+											},
+											Key: "OPEN_API_KEY",
```
Comment (priyanka-operant) [inline internal/components/components.go:137]: and here

**Review by priyanka-operant — APPROVED** [review]

**Comment by priyanka-operant** [comment]
i've been adding issues and connecting PRs with them as I go

**Comment by glenn-operant** [comment]
Closes #60

**Pull request merged**